ca/ca_test.go

```diff
@@ -0,0 +1,189 @@
+package ca
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/bwesterb/mtc"
+)
+
+// testCA is a Merkle Tree CA that runs in memory.
+type testCA struct {
+	*Handle
+	path string
+}
+
+// newTestCA creates a CA in a temporary directory. The caller should call
+// `Close()` in order to delete the directory once the test is finished.
+func newTestCA(t *testing.T) *testCA {
+	path, err := os.MkdirTemp("", "test-mtca-*")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	issuer := mtc.RelativeOID{}
+	err = issuer.UnmarshalText([]byte("1.2.3.4"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	handle, err := New(path, NewOpts{
+		Issuer:        issuer,
+		ServerPrefix:  "ca.example.com",
+		BatchDuration: time.Duration(0),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return &testCA{
+		handle,
+		path,
+	}
+}
+
+func (ca *testCA) Close() {
+	ca.Handle.Close()
+	if err := os.RemoveAll(ca.path); err != nil {
+		panic(err)
+	}
+}
+
+func createVerifyTest(t *testing.T) (*mtc.BikeshedCertificate,
+	*mtc.SignedValidityWindow, mtc.CAParams) {
+	ca := newTestCA(t)
+	defer ca.Close()
+
+	sk, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	subject, err := mtc.NewTLSSubject(
+		mtc.TLSECDSAWithP256AndSHA256,
+		sk.Public(),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assertion := mtc.Assertion{
+		Subject: subject,
+		Claims: mtc.Claims{
+			DNS: []string{"example.org"},
+		},
+	}
+
+	notAfter := time.Now().Add(time.Hour * 24 * 7)
+
+	assertionRequest := &mtc.AssertionRequest{
+		Assertion: assertion,
+		NotAfter:  notAfter,
```

Comment:
This parameter should be optional fwiw--it's just if the requester wants an even shorter lifetime than the batch lifetime.
```suggestion
		NotAfter:  notAfter,
```

Reply:
I'd probably omit it from these tests, since we're not trying to test that specific behavior.

Reply:
We check the zero value in ca.go: https://github.com/bwesterb/mtc/blob/main/ca/ca.go#L166

Reply:
Done.